python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -936,24 +936,22 @@ def display2dArray(arr, title='Data', extent=None):
         bbox = footprint.getBBox()
         extent = (bbox.getBeginX(), bbox.getEndX(), bbox.getBeginY(), bbox.getEndY())
         if z.shape[0] == 3:
-            fig = plt.figure(figsize=(8, 8))
+            plt.figure(figsize=(8, 8))
             for i in range(3):
                 plt.subplot(3, 3, i*3+1)
                 display2dArray(z[i, :], 'Data', extent=extent)
                 plt.subplot(3, 3, i*3+2)
                 display2dArray(fit[i, :], 'Model', extent=extent)
                 plt.subplot(3, 3, i*3+3)
                 display2dArray(z[i, :] - fit[i, :], 'Residual', extent=extent)
-            return fig
         else:
-            fig = plt.figure(figsize=(8, 2.5))
+            plt.figure(figsize=(8, 2.5))
             plt.subplot(1, 3, 1)
             display2dArray(z, 'Data', extent=extent)
             plt.subplot(1, 3, 2)
             display2dArray(fit, 'Model', extent=extent)
             plt.subplot(1, 3, 3)
             display2dArray(z - fit, 'Residual', extent=extent)
-            return fig
```

Comment:
Why remove `fig` from the return? Note that it is still documented as the return value. Also the function `display2dArray` does still return `fig`, but that isn't used.

Reply:
Because it was causing the `show` down below to not get used, and it wasn't used by anything.